homeassistant/components/light/rflink.py

```diff
@@ -170,12 +145,9 @@ def devices_from_config(domain_config, hass=None):
         device = entity_class(device_id, hass, **device_config)
         async_add_entities([device])
 
-        # Register entity to listen to incoming Rflink events
-        hass.data[DATA_ENTITY_LOOKUP][
-            EVENT_KEY_COMMAND][device_id].append(device)
-
         # Schedule task to process event after entity is created
-        hass.async_add_job(device.handle_event, event)
+        hass.async_add_job(device.handle_event_callback,
```

Comment:
Send a signal instead.

Reply:
Fixed.

Reply:
Ok, I guess we can't guarantee that the signal has been connected even if adding the sending of the signal to the job queue.

So keep this as is for now. It would be nice if there's a way to do it with signaling so we always use the same interface on the entity for the event processing.

Reply:
Maybe we pass in the first event to the entity when we create the entity? The entity stores the first event as an instance attribute and process it in `async_added_to_hass`.

Reply:
OK